testing/test_capture.py

```diff
@@ -668,6 +668,20 @@ def test_a(fix):
         reprec = testdir.inline_run()
         reprec.assertoutcome(passed=1)
 
+    def test_keeps_sys_fd_encoding(self, testdir, monkeypatch):
+        monkeypatch.setenv("LC_ALL", "C")
+        testdir.makepyfile(
+            """
+            def test_encoding():
+                import sys
+                assert sys.stdout.encoding == "ANSI_X3.4-1968"
+            """
+        )
+        result = testdir.runpytest_subprocess("-s")
+        result.stdout.fnmatch_lines(["*1 passed in*"])
+        result = testdir.runpytest_subprocess()
+        result.stdout.fnmatch_lines(["*1 passed in*"])
+
```

Comment:
Is there a better way to test this?

Unit test appears to be problematic, since the outer pytest might be wrong already, but two subprocess tests are maybe overkill?